packages/nx/src/plugins/js/project-graph/build-dependencies/explicit-package-json-dependencies.ts

```diff
@@ -80,29 +92,61 @@ function processPackageJson(
         };
         validateDependency(dependency, ctx);
         collectedDeps.push(dependency);
-      } else if (ctx.externalNodes[`npm:${d}`]) {
-        const dependency: RawProjectGraphDependency = {
-          source: sourceProject,
-          target: `npm:${d}`,
-          sourceFile: fileName,
-          type: DependencyType.static,
-        };
-        validateDependency(dependency, ctx);
-        collectedDeps.push(dependency);
+        continue;
       }
-    });
+
+      const targetProjectLocator = new TargetProjectLocator(
```

Comment:
We shouldn't re-create this every single `package.json`.

Create it once in `buildExplicitPackageJsonDependencies`